fetch-in-react-practice/client/components/app.jsx

```diff
@@ -39,21 +39,32 @@ export default class App extends React.Component {
     * TIP: Use Array.prototype.concat to create a new array containing the contents
     * of the old array, plus the object returned by the server.
     */
-
+    const todos = this.state.todos;
     fetch('http://localhost:3000/api/todos', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json'
       },
       body: JSON.stringify(newTodo)
     })
-      .then(res => res.json());
+      .then(res => res.json())
+      .then(data => {
+        todos.push(data);
+        this.setState({ todos: todos });
+      });
 
   }
 
   toggleCompleted(todoId) {
 
-    const newCompleted = this.state.todos[todoId - 1];
+    const allState = this.state.todos;
+    let newCompleted;
+    for (let i = 0; i < allState.length; i++) {
+      if (allState[i].todoId === todoId) {
+        newCompleted = allState[i];
+      }
+    }
+
     let newIsComplete = {};
     if (newCompleted.isCompleted) {
       newCompleted.isCompleted = false;
```

Comment:
This is modifying your original object from the state. try to make a copy first.